libp2p/routing_record.nim

```diff
@@ -0,0 +1,125 @@
+## Nim-Libp2p
+## Copyright (c) 2021 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+## This module implements Routing Records.
+
+{.push raises: [Defect].}
+
+import std/[sequtils, times]
+import pkg/stew/[results, byteutils]
+import
+  multiaddress,
+  multicodec,
+  peerid,
+  protobuf/minprotobuf,
+  signed_envelope
+
+export peerid, multiaddress, signed_envelope
+
+## Constants relating to signed peer records
+const
+  EnvelopeDomain = multiCodec("libp2p-peer-record") # envelope domain as per RFC0002
+  EnvelopePayloadType= @[(byte) 0x03, (byte) 0x01] # payload_type for routing records as spec'ed in RFC0003
```

Comment:
This should be initialized with the appropriate multicodec, since it's already in our multicodec table.

Reply:
See in https://github.com/status-im/nim-libp2p/pull/649#discussion_r748341428
AFAICt there is no good way to extract only 0x03 0x01 (without padding) from our multiCodec implementation, and we have to use exactly that to be compatible with go-libp2p
Not sure if it's something we are missing, or if they should have padding in their value

Reply:
I don't think thats the case, `multiCodec()` will return the raw codec, not the length prefixed one. Codecs are not length prefixed, things like `Cid`, `multihash`, etc... are.

```nim
const
  CodeCodecs = initMultiCodecCodeTable()
  NameCodecs = initMultiCodecNameTable()

proc multiCodec*(name: string): MultiCodec {.compileTime.} =
  ## Generate MultiCodec from string ``name`` at compile time.
  let code = NameCodecs.getOrDefault(name, -1)
  doAssert(code != -1)
  MultiCodec(code)

proc multiCodec*(code: int): MultiCodec {.compileTime.} =
  ## Generate MultiCodec from integer ``code`` at compile time.
  let name = CodeCodecs.getOrDefault(code, "")
  doAssert(name != "")
  MultiCodec(code)
```